tests/hla/process_randomlist.py

```diff
@@ -0,0 +1,204 @@
+        aligned to an astrometric standard.
+    """
+
+    def test_align_randomFields(self):
+        """ Wrapper to set up the test for aligning a large number of randomly
+            selected fields (aka datasets) from a input ascii file (CSV).
+
+            The wrapper provides the parameter settings for the underlying test,
+            as well as implements the criterion for the overall success or failure
+            of the test.
+        """
+
+        inputListFile = 'ACSWFC3List.csv'
+
+        # Desired number of random entries for testing
+        inputNumEntries = 50
+
+        # Seed for random number generator
+        inputSeedValue = 1
+
+        # Obtain the full path to the file containing the dataset field names
+        self.input_repo = 'hst-hla-pipeline'
+        self.tree = 'dev'
+        self.input_loc = 'master_lists'
+        input_file_path = self.get_data(inputListFile)
+
+        # Randomly select a subset of field names (each field represented by a row) from
+        # the master CSV file and return as an Astropy table
+        randomCandidateTable = catutils.randomSelectFromCSV(input_file_path[0],
+            inputNumEntries, inputSeedValue)
+
+        # Invoke the methods which will handle acquiring/downloading the data from
+        # MAST and perform the alignment
+        percentSuccess = 0.0
+        try:
+            percentSuccess = self.align_randomFields (randomCandidateTable)
+        except Exception:
+            pass
+
+        return(percentSuccess)
+
+    def align_randomFields(self, randomTable):
+        """ Process randomly selected fields (aka datasets) stored in an Astropy table.
+
+            Each field is used as input to determine if it can be aligned to an
+            astrometric standard.  The success or fail status for each test is retained
+            as the overall success or fail statistic is the necessary output from
+            this test.
+        """
+
+        numSuccess = 0
+        numQualSuccess = 0
+        numUnsuccessful = 0
+        numException = 0
+        numProcessedDatasets = 0
+
+        # Read the table and extract a list of each dataset name in IPPSSOOT format
+        # which is either an association ID or an individual filename
+        dataset_list = get_dataset_list(randomTable)
+
+        numProcessedDatasets = len(dataset_list)
+        numStartTests  = numProcessedDatasets
+
+        # Process the dataset names in the list
+        #
+        # If the dataset name represents an association ID, the multiplicity
+        # of images within the association need to be processed.  Otherwise,
+        # the dataset is a single image.
+        #
+        # If the "alignment" of a field/dataset fails for any reason, trap
+        # the exception and keep going.
+        allDatasetTable = Table()
+        datasetKey = -1
+        print("TEST_RANDOM. Dataset List: ", dataset_list)
+        for dataset in dataset_list:
```

Comment:
This is one instance where you could use `enumerate(dataset_list)` to define `datasetKey` automatically, rather than manually incrementing a separately defined variable.  No change in function, but perhaps less room for mistakes in incrementing this variable.  This suggestion, though, would not be appropriate if additional logic was being applied to incrementing this variable, but I could not find any in this specific method.

Reply:
Point taken though I am going to leave this alone as this variable (and the column in the output table) seems to be on its way to being deprecated.